packages/code-snippet/src/CodeSnippetWidget.tsx

```diff
@@ -37,20 +55,153 @@ const COPY_ICON_CLASS = 'elyra-copy-icon';
+        // ISSUE: This is not being triggered by a PythonFileEditor
+        //TODO remove debug line below
+        console.log('This is a PythonFileEditor');
+
+        this.verifyLanguageAndInsert(snippet, 'python', fileEditor);
+      } else {
+        fileEditor.replaceSelection(snippetStr);
+      }
+    } else if (widget instanceof NotebookPanel) {
+      const notebookWidget = widget as NotebookPanel;
+      const notebookCellEditor = (notebookWidget.content as Notebook).activeCell
+        .editor;
+
+      // Original approach:
+      // ISSUE: kernelLanguage is an empty string, not updated until a new notebook is saved, closed and reopened
+      // const kernelLanguage: string =
+      //   notebookWidget.context.sessionContext.kernelPreference.language;
+
+      // GET NOTEBOOK KERNEL SPEC
+      // This is now handled before a new notebook is saved
+      // ISSUE: it might take some time such as a spark cluster
+      const kernelInfo = await notebookWidget.sessionContext.session?.kernel
+        ?.info;
+      const kernelLanguage: string = kernelInfo?.language_info.name || '';
+      console.log('KERNEL LANG: ' + kernelLanguage);
+      this.verifyLanguageAndInsert(snippet, kernelLanguage, notebookCellEditor);
+
+      // ISSUE: kernel name doesn't define kernel language, they might differ in some cases
+      // const kernelLanguage: string = notebookWidget.sessionContext.session?.kernel?.name;
+      // console.log('KERNEL LANG: ' + kernelLanguage);
+      // this.verifyLanguageAndInsert(snippet, kernelLanguage, notebookCellEditor);
+
+      // const kernelManager = new KernelManager();
+      // const specsManager = new KernelSpecManager();
+      // const sessionManager = new SessionManager({ kernelManager });
+      // const sessionContext = new SessionContext({
+      //   sessionManager,
+      //   specsManager,
+      //   name: 'Example'
+      // });
+      //
+      // // This never gets called
+      // sessionContext.kernelChanged.connect(() => {
+      //     void sessionContext.session?.kernel?.info.then(info => {
+      //       const lang = info.language_info;
+      //       console.log('KERNEL LANG: '+ lang);
+      //     });
+      //   });
+    } else {
+      this.showErrDialog('Code snippet insert failed: Unsupported widget');
+    }
+  };
+
+  // Handle language compatibility between code snippet and editor
+  private verifyLanguageAndInsert = async (
+    snippet: ICodeSnippet,
+    editorLanguage: string,
+    editor: CodeEditor.IEditor
+  ): Promise<void> => {
+    console.log(
+      'editorLanguage: ' +
+        editorLanguage +
+        ', snippet.languange: ' +
+        snippet.language
+    );
+    const snippetStr: string = snippet.code.join('\n');
+    if (editorLanguage === '') {
+      // Allow code snippet insertion when editorLanguage is unavailable
+      editor.replaceSelection(snippetStr);
+    } else if (snippet.language !== editorLanguage) {
```

Comment:
rather than having an else if here we can combine the first two checks
```suggestion
    if (editorLanguage && snippet.language !== editorLanguage) {
```